interfaces/builtin/avahi_control.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/snapcore/snapd/release"
+)
+
+const avahiControlBaseDeclarationSlots = `
+  avahi-control:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+const avahiControlSummary = `allows control over service discovery on a local network via the mDNS/DNS-SD protocol suite`
+
+const avahiControlConnectedSlotAppArmor = `
+# EntryGroup
+dbus (receive)
+    bus=system
+    path=/Client*/EntryGroup*
+    interface=org.freedesktop.Avahi.EntryGroup
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=system
+    interface=org.freedesktop.Avahi.EntryGroup
+    member=StateChanged
+    peer=(name=org.freedesktop.Avahi, label=###PLUG_SECURITY_TAGS###),
+`
+
+const avahiControlConnectedPlugAppArmor = `
+dbus (send)
+    bus=system
+    path=/
+    interface=org.freedesktop.Avahi.Server
+    member=Set*
+    peer=(name=org.freedesktop.Avahi,label=###SLOT_SECURITY_TAGS###),
+
+# EntryGroup
+dbus (send)
+    bus=system
+    path=/
+    interface=org.freedesktop.Avahi.Server
+    member=EntryGroupNew
+    peer=(name=org.freedesktop.Avahi, label=###SLOT_SECURITY_TAGS###),
+
+dbus (send)
+    bus=system
+    path=/Client*/EntryGroup*
+    interface=org.freedesktop.Avahi.EntryGroup
+    member={Free,Commit,Reset}
+    peer=(name=org.freedesktop.Avahi, label=###SLOT_SECURITY_TAGS###),
+
+dbus (send)
+    bus=system
+    path=/Client*/EntryGroup*
+    interface=org.freedesktop.Avahi.EntryGroup
+    member={GetState,IsEmpty,UpdateServiceTxt}
+    peer=(name=org.freedesktop.Avahi, label=###SLOT_SECURITY_TAGS###),
+
+dbus (send)
+    bus=system
+    path=/Client*/EntryGroup*
+    interface=org.freedesktop.Avahi.EntryGroup
+    member=Add{Service,ServiceSubtype,Address,Record}
+    peer=(name=org.freedesktop.Avahi, label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=system
+    path=/Client*/EntryGroup*
+    interface=org.freedesktop.Avahi.EntryGroup
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+type avahiControlInterface struct{}
+
+func (iface *avahiControlInterface) Name() string {
+	return "avahi-control"
+}
+
+func (iface *avahiControlInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              avahiControlSummary,
+		ImplicitOnClassic:    true,
+		BaseDeclarationSlots: avahiControlBaseDeclarationSlots,
+	}
+}
+
+func (iface *avahiControlInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.Plug, plugAttrs map[string]interface{}, slot *interfaces.Slot, slotAttrs map[string]interface{}) error {
+	old := "###SLOT_SECURITY_TAGS###"
+	var new string
+	if release.OnClassic {
+		// If we're running on classic Avahi will be part
+		// of the OS snap and will run unconfined.
+		new = "unconfined"
+	} else {
+		new = slotAppLabelExpr(slot)
+	}
+	snippet := strings.Replace(avahiObserveConnectedPlugAppArmor, old, new, -1)
```

Comment:
I think it is fine to re-use this variable, but can you make a comment here. I suggest:
```
# avahi-control implies avahi-observe, so add snippets for both here
```